### PR TITLE
feat: migrate cloudrunner internals to slog

### PR DESCRIPTION
**Pull request by odsod**
This is the turning-point for the migration from zap to slog - all
internal logging in cloudrunner now uses the slog APIs and the slog
logging backends (TextHandler in development and JSONHandler in
production).

The zap APIs continue to work, and will be marked as deprecated when the
slog APIs have proven stable in production for a few weeks.

### CODE REVIEW

File: cloudtrace/middleware.go [hunk 1]
```diff
@@ -89,22 +84,3 @@ func (i *Middleware) withInternalContext(ctx context.Context, header string) con
 	}
 	return SetContext(ctx, result)
 }
-
-func (i *Middleware) withLogTracing(ctx context.Context, header string) context.Context {
```
Comment (odsod) [inline cloudtrace/middleware.go:93]: FYI This is handeld automatically by the cloudrunner slog logging handler - trace context is appended on every log line.

File: run.go [hunk 2]
```diff
@@ -191,35 +192,3 @@ func getRunContext(ctx context.Context) (*runContext, bool) {
 	result, ok := ctx.Value(runContextKey{}).(*runContext)
 	return result, ok
 }
-
-type buildInfoMarshaler struct {
```
Comment (odsod) [inline run.go:195]: This is moved to the slog handler.

File: cloudzap/level.go [hunk 3]
```diff
@@ -24,3 +28,19 @@ func LevelToSeverity(l zapcore.Level) string {
 		return "DEFAULT"
 	}
 }
+
+// LevelToSlog converts a [zapcore.Level] to an [slog.Level].
```
Comment (quoral) [inline cloudzap/level.go:32]:
```suggestion
// LevelToSlog converts a [zapcore.Level] to a [slog.Level].
```

**Review by quoral — APPROVED** [review]
Looks super reasonable!

**Pull request merged**